### PR TITLE
http response: check if result would fit into chunkLen

### PR DESCRIPTION
**Pull request by vmiklos**
Signed-off-by: Miklos Vajna <vmiklos@collabora.com>
Change-Id: I6e3b73461653d2ab3cedaa3f6ca7fbfd2a826edb
(cherry picked from commit 57d3d0f3919115b77761ae26af4ff40140fb7f98)

### CODE REVIEW

**Comment by vmiklos** [comment]
@Ashod could you please take a look? Thanks.

**Comment by vmiklos** [comment]
> 09:34 < kendy> @vmiklos: Given that "digit" is always less than 16, you can avoid the "- digit" in the condition (because the division turns that into 0 anyway), making it constexpr, can't you?

Done.

**Comment by vmiklos** [comment]
unit-wopi-ownertermination failed, retry.

**Comment by vmiklos** [comment]
unit-wopi-ownertermination again, rebase.

File: net/HttpRequest.cpp [hunk 1]
```diff
@@ -553,6 +553,12 @@ int64_t Response::readData(const char* p, int64_t len)
                     if (digit < 0)
                         break;
 
+                    // Can assume that digit is always less than 16.
+                    if (chunkLen >= std::numeric_limits<int64_t>::max() / 16)
+                    {
+                        // Would not fit into chunkLen.
+                        return len - available;
```
Comment (Ashod) [inline net/HttpRequest.cpp:560]: Wouldn't this be a fatal error? Would be be able to recover? If not, best to return -1 and fail.

**Review by Ashod — COMMENTED** [review]
Thanks Miklos!

File: net/HttpRequest.cpp (hunk 1 above)
Comment (vmiklos) [reply net/HttpRequest.cpp:560]: Good point, -1 is better here, this is really a garbage input, we just should not crash on this.

**Comment by vmiklos** [comment]
@Ashod OK CI is now happy with the 3rd commit, addressing your review. Please let me know if you need anything more from me. Thanks!

**Review by Ashod — APPROVED** [review]
Thanks Miklos, looks good to me.

**Comment by Ashod** [comment]
@vmiklos, do we want to squash the 3 commits before merging?

**Comment by vmiklos** [comment]
I would leave them separate, otherwise my life is harder when I try to make sure all 3 are in all relevant branches. :-)

**Pull request merged**